common/VotingUtils.js

```diff
@@ -177,7 +177,24 @@ const batchRetrieveRewards = async (requests, roundId, votingContract, account)
   };
 };
 
+// Get the latest event matching the provided parameters. Assumes that all events from Voting.sol have indexed
+// parameters for identifier, roundId, and voter.
+const getLatestEvent = async (eventName, request, roundId, account, votingContract) => {
+  const events = await votingContract.getPastEvents(eventName, {
+    fromBlock: 0,
+    filter: { identifier: request.identifier, roundId: roundId.toString(), voter: account.toString() }
+  });
+  let matchedEvent = null;
+  for (const ev of events) {
+    if (ev.returnValues.time.toString() === request.time.toString()) {
+      matchedEvent = ev.returnValues;
+    }
```

Comment:
This may be a dumb question, but are the returned events guaranteed to be sorted?

Reply:
They seem to be, but it's not documented. I copied your explicit sort function (which at some point we could/should unify).

Reply:
@mrice32 If we sort descending, we can exit early. PTAL.